addons/account/models/res_partner_bank.py

```diff
@@ -253,6 +253,11 @@ def create(self, vals_list):
                 # force the allow_out_payment field to False in order to prevent scam payments on newly created bank accounts
                 vals['allow_out_payment'] = False
 
+        for vals in vals_list:
+            archived_res_partner_bank = self.env['res.partner.bank'].search([('active', '=', False), ('partner_id', '=', vals['partner_id']), ('acc_number', '=', vals['acc_number'])])
+            if archived_res_partner_bank:
+                raise UserError(_("A bank account with Account Number %s already exists for Partner %s, but is archived. Please unarchive it instead.", vals['acc_number'], vals['acc_holder_name']))
```

Comment:
this will only fix the create, when the issue also happen with the write

it would be better in an @api.constrains, but I guess we can't because we want to act before the sql constraint.

so could be ok because I don't see another way.

also acc_number, acc_holder_name, active, partner_ don't seem to be required field, so it seems that we could call the method without them, we probably would need to have a if and check them with .get

Reply:
```suggestion
                raise UserError(_("A bank account with Account Number %(number)s already exists for Partner %(partner)s, but is archived. Please unarchive it instead.", number=vals['acc_number'], partner=vals['acc_holder_name']))
```
If you name the parameters, it will allow translators to switch the ordering for languages where that would make more sense.